lib/mongoose-store.js

```diff
@@ -45,13 +57,20 @@ Store.prototype.findById = function(id) {
 };
 
 Store.prototype.read = function(_id) {
-  return this.model.findOne({id: _id}).exec().then(convertToJSON);
+  return this.model.findOne({id: _id}).exec().then(function(foundDocument) {
+
+    if (!foundDocument) {
+      return createNoDocumentError(_id);
+    }
+
+    return foundDocument;
+  }).then(convertToJSON);
 };
 
 Store.prototype.update = function(object) {
   return this.model.findOne({id: object.id}).exec().then(function(foundDocument) {
     if (!foundDocument) {
-      return Promise.reject(new Error("No document with id " + object.id  + " found"));
+      return createNoDocumentError(object.id);
```

Comment:
I don't think so. If someone is trying to update something that isn't there it should be an error.